manifests/charts/istio-control/istio-discovery/files/gateway-injection-template.yaml

```diff
@@ -192,16 +192,16 @@ spec:
     - name: istio-podinfo
       mountPath: /etc/istio/pod
   volumes:
-  - emptyDir: {}
```

Comment:
Why change this? I doubt it matters but it seems less clear and like the thing that could cause weird issues with some tools

Reply:
Actually I don't get it; how does this change fix spire?

Reply:
Specifically, it would be the CSI driver, not SPIRE persay. I don't think it has anything to do with SPIRE or the CSI driver, I think it mostly has to do with our template overlay logic working differently for 

`emptyDir:` vs `emptyDir:{}`

We are wildly inconsistent with this, to the PR author's point, we use `emptyDir:` in the injection template and have for some time, and use a mishmash of `emptyDir:` and `emptyDir: {}` everywhere else.

Reply:
In [our SPIRE integration guide we use overlays for gateways](https://istio.io/latest/docs/ops/integrations/spire/#install-istio), not injection templates.

Reply:
> (preferred format for the answer is a unit test 🙂 )

Agree. That being said, this is not an invalid change and it's a reasonable quick fix - the templates should be consistent with each other on this point, and to your point the YAML should be equivalent - if not, probably a bug in our templating logic which can be addressed separately.

Reply:
Honestly, I don’t know how it fixes the issue, but it has fixed it in my case. And it doesn’t seem like it could break anything (speaking cautiously here, because I can’t be certain, though). I’d be glad to add a unit test, but could you give me hint on how to do so?